test/unit/date/parse.js

```diff
@@ -471,6 +471,12 @@ QUnit.test( "should parse milliseconds in a day (A+)", function( assert ) {
 	assertParse( assert, "6330736900", "AAAAA", cldr, date1 );
 });
 
+QUnit.test( "should allow strict parsing of the date", function( assert ) {
+	date1 = new Date(2001,0,1);
+	assertParse( assert, "1/1/01", "M/d/yy", cldr, date1, true );
+	assertParse( assert, "1/1/2001", "M/d/yy", cldr, null, true );
+});
```

Comment:
Let's include `strict` tests for additional fields. I have tested it for `M`, but it falsely passes.

```js
assertParse( assert, "01/1/01", "M/d/yy", cldr, null, true );
// Should return null because of the padded month, but it returns date1.
```

So:

 1. Fix solution for `M` case above.
 1. What other fields can we test for? Only numerical? Numerical timezones? Any non-numerical ones?